mhs/mhs/common/state/dynamo_persistence_adaptor.py

```diff
@@ -10,57 +11,47 @@
 
 
 class RecordCreationError(RuntimeError):
+    """Error occurred when creating record."""
     pass
 
 
 class RecordDeletionError(RuntimeError):
+    """Error occurred when deleting record."""
     pass
 
 
 class RecordRetrievalError(RuntimeError):
-    pass
-
-
-class InvalidTableError(RuntimeError):
+    """Error occurred when retrieving record."""
     pass
 
 
 class DynamoPersistenceAdaptor(mhs.common.state.persistence_adaptor.PersistenceAdaptor):
-    """
-    Class responsible for persisting items into DynamoDB.
-    """
+    """Class responsible for persisting items into a DynamoDB."""
 
     def __init__(self, **kwargs):
         """
         Constructs a DynamoDB version of a
         :class:`PersistenceAdaptor <mhs.common.state.persistence_adaptor.PersistenceAdaptor>`.
         The kwargs provided should contain the following information:
-          * state_table: The name used to refer to the table containing state items.
-          * state_arn: The Amazon Resource Name used to identify the dynamo table containing state items.
-          * sync_async_table: The name used to refer to the table containing sync_async message items.
-          * sync_async_arn: The Amazon Resource Name used to identify the dynamo table containing sync_async message
-            items.
+          * table_name: The Table Name used to identify the dynamo table containing required items.
         :param kwargs: The key word arguments required for this constructor.
         """
-        state_table = kwargs.get('state_table', 'state')
-        state_arn = kwargs.get('state_arn')
-        sync_async_table = kwargs.get('sync_async_table', 'sync_async')
-        sync_async_arn = kwargs.get('sync_async_arn')
+        self.table_name = kwargs.get('table_name')
 
-        dynamo_db = aioboto3.resource('dynamodb')
+    async def add(self, key, data):
+        """Add an item to a specified table, using a provided key.
 
-        self.tables = {
-            state_table: dynamo_db.Table(state_arn),
-            sync_async_table: dynamo_db.Table(sync_async_arn)
-        }
-
-    async def add(self, table_name, key, item):
-        logger.info('011', 'Adding {record} for {key}', {'record': item, 'key': key})
+        :param key: The key used to identify the item.
+        :param data: The item to store in persistence.
+        :return: The previous version of the item which has been replaced. (None if no previous item)
+        """
+        logger.info('011', 'Adding {record} for {key}', {'record': data, 'key': key})
         try:
-            response = await self._resolve_table(table_name).put_item(
-                Item={'key': key, 'data': json.dumps(item)},
-                ReturnValues='ALL_OLD'
-            )
+            async with aioboto3.resource('dynamodb', region_name='eu-west-2') as dynamo_resource:
```

Comment:
Extract common string. Actually, a better way is to extract the common with invocation. I think this can be done something like:
```python
@contextlib.asynccontextmanager
async def create_dynamo_table(self):
    async with aioboto3.resource('dynamodb', region_name='eu-west-2') as dynamo_resource:
        yield dynamo_resource.Table(self.table_name)
```
Possibly in here, you can log the table that's being used as well.

Reply:
Done.